### ISSUE TITLE
Make images fill slide without using background-image

### ISSUE THREAD

**Issue by williamsmj** [gnab/remark#331]
Is it possible to scale an image to the largest size such that both its width and its height can fit inside the slide?

Right now I am defining a css class like below for every slide I want to do this in, which is the kind of thing I was hoping to avoid by using markdown!

``` css
.foo {
    background-image: url(foo.jpg);
    background-size: contain;
}
```

**Comment by geewiz** [comment]
Have you tried using the `max-width` and `max-height` CSS properties?

**Comment by tripu** [comment]
I think the title of this issue is misleading; it has little to do with Remark. (You don't need to set the image as a background: you can also tweak an image's properties to fill the whole width/height, give it a different `z-index`, etc.)

I guess what you want to know is whether there's a [predefined content class](https://github.com/gnab/remark/wiki/Markdown#content-classes) that does exactly what you need?

**Comment by Tschis** [comment]
I don't think the title is misleading because markdown automatically scales the image to fit the screen.
It does not happen when using remark.

**Comment by tripu** [comment]
> *&ldquo;Markdown automatically scales the image to fit the screen.&rdquo;*

Sorry, what do you mean?

An image in Markdown will produce an `<img>` in HTML; there's nothing implicit about how that image's going to by styled&hellip;

**Comment by Tschis** [comment]
@tripu 

If you use this as an example markdown:

```
# Intent
![Hexagonal Architecture class diagram](https://raw.githubusercontent.com/iluwatar/java-design-patterns/master/hexagonal/etc/hexagonal.png)
```

You will notice it produces an image that is scaled and fits the screen.

However, if you throw this same .md file into [remarkise](https://remarkjs.com/remarkise) or just build your remark presentation, you will notice the image does not fit the slide.

**Comment by tripu** [comment]
@Tschis:

> *&ldquo;It produces an image that is scaled and fits the screen.&rdquo;*

No, it doesn't.
See [this GitHub gist](https://gist.github.com/tripu/45586863859dd90bbe0bb903dbaac6af). The second image is not scaled, but shown in its original size.

> *&ldquo;You can see in the generated HTML [of this .md](https://github.com/iluwatar/java-design-patterns/blob/master/hexagonal/README.md) that there is a `style="max-width:100%;"`&rdquo;*

That's just GitHub's own styling of its content. Nothing to do with Markdown [image syntax in [CommonMark](http://spec.commonmark.org/0.28/#images); in [Daring Fireball](https://daringfireball.net/projects/markdown/syntax#img)]. It's GH doing something additional, not Remark lacking a feature of Markdown.

**Comment by tripu** [comment]
@williamsmj, this syntax is a bit more concise and will give you what you want (I think):

```
<style>
  .remark-slide-content {
    background-size: contain;
  }
</style>

background-image: url("foo.jpeg")
# Slide 1

---

background-image: url("bar.jpeg")
# Slide 2

---

etc
```

At least this avoids repeating `background-size: contain;` once for every slide, and defining new classes just for this&hellip;

See an example: [rendered](https://tripu.github.io/remark/remarkise?url=https%3A%2F%2Ftripu.github.io%2FEvents%2F2014-11-28_Granada-Spain_UGR_W3C-TTWF%2Fpresentation.md#2), [source](https://github.com/tripu/Events/blob/master/2014-11-28_Granada-Spain_UGR_W3C-TTWF/presentation.md).

**Comment by Tschis** [comment]
@tripu 
Sorry, with "scaled" I meant "scaled down" to be able to be completely shown on screen.

I also used [MarkDownLivePreview](http://markdownlivepreview.com) and [Dillinger](https://dillinger.io) to test my example code, and both of them showed the image fully on screen, so I figured that's how it was supposed to work. Rechecking their generated HTML now shows no sign of the max-width being included, so I stand corrected. Still, somehow they fit the image within the screen.

@williamsmj I'm not entirely sure if your problem is the same as mine. But for what it's worth, I solved my problem by just adding the code below to the `<style>` of my remark file:
```
	  img {
		max-width:100%;
	  }
```

And then just using images normally `![Image](image.png)`
